Generated typings include implicit any

If the typescript compiler has the flag `noImplicitAny` enabled, some errors appear. These errors were introduced here https://github.com/octokit/rest.js/pull/1113

```
> tsc -p ./

node_modules/@octokit/rest/index.d.ts(22759,5): error TS7010: 'before', which lacks return-type annotation, implicitly has an 'any' return type.
node_modules/@octokit/rest/index.d.ts(22760,5): error TS7010: 'after', which lacks return-type annotation, implicitly has an 'any' return type.
node_modules/@octokit/rest/index.d.ts(22767,5): error TS7010: 'error', which lacks return-type annotation, implicitly has an 'any' return type.
node_modules/@octokit/rest/index.d.ts(22771,5): error TS7010: 'wrap', which lacks return-type annotation, implicitly has an 'any' return type.
node_modules/@octokit/rest/index.d.ts(22780,3): error TS7010: 'plugin', which lacks return-type annotation, implicitly has an 'any' return type.
node_modules/@octokit/rest/index.d.ts(22782,3): error TS7010: 'registerEndpoints', which lacks return-type annotation, implicitly has an 'any' return type.

```

As far as I could see, all the hooks and `registerEndpoints` are `void`, and `plugin` returns a `Github` instance.

--- a/package.json
+++ b/package.json
@@ -101,7 +101,7 @@
     "generate-bundle-report": "webpack-bundle-analyzer dist/bundle-stats.json --mode=static --no-open --report dist/bundle-report.html",
     "generate-routes": "node scripts/generate-routes",
     "prevalidate:ts": "npm run -s build:ts",
-    "validate:ts": "tsc --target es6 index.d.ts",
+    "validate:ts": "tsc --target es6 --noImplicitAny index.d.ts",
     "postvalidate:ts": "tsc --noEmit --target es6 test/typescript-validate.ts",
     "deploy-docs": "gh-pages-with-token -d doc",
     "semantic-release": "semantic-release",

--- a/scripts/lib/get-routes.js
+++ b/scripts/lib/get-routes.js
@@ -54,7 +54,14 @@ function getRoutes () {
         indexes.forEach(i => endpoint.params.splice(i, 1))
         endpoint.params.unshift(
           Object.assign(contentLengthParam, { name: 'headers.content-length' }),
-          Object.assign(contentTypeParam, { name: 'headers.content-type' })
+          Object.assign(contentTypeParam, { name: 'headers.content-type' }),
+          {
+            name: 'headers',
+            location: 'headers',
+            required: true,
+            type: 'object',
+            description: 'Request headers containing `content-type` and `content-length`'
+          }
         )
       }
 

--- a/scripts/templates/index.d.ts.tpl
+++ b/scripts/templates/index.d.ts.tpl
@@ -201,15 +201,15 @@ declare class Github {
   authenticate(auth: Github.Auth): void;
 
   hook: {
-    before(name: string, callback: (options: Github.HookOptions) => void)
-    after(name: string, callback: (response: Github.Response<any>, options: Github.HookOptions) => void)
-    error(name: string, callback: (error: Github.HookError, options: Github.HookOptions) => void)
-    wrap(name: string, callback: (request: (options: Github.HookOptions) => Promise<Github.Response<any>>, options: Github.HookOptions) => void)
+    before(name: string, callback: (options: Github.HookOptions) => void): void
+    after(name: string, callback: (response: Github.Response<any>, options: Github.HookOptions) => void): void
+    error(name: string, callback: (error: Github.HookError, options: Github.HookOptions) => void): void
+    wrap(name: string, callback: (request: (options: Github.HookOptions) => Promise<Github.Response<any>>, options: Github.HookOptions) => void): void
   }
 
-  plugin(plugin: Github.Plugin | [Github.Plugin])
+  plugin(plugin: Github.Plugin | [Github.Plugin]): Github
 
-  registerEndpoints(routes: any)
+  registerEndpoints(routes: any): void
 
   request: any
 

--- a/scripts/templates/index.d.ts.tpl
+++ b/scripts/templates/index.d.ts.tpl
@@ -207,7 +207,7 @@ declare class Github {
     wrap(name: string, callback: (request: (options: Github.HookOptions) => Promise<Github.Response<any>>, options: Github.HookOptions) => void): void
   }
 
-  plugin(plugin: Github.Plugin | [Github.Plugin]): Github
+  static plugin(plugin: Github.Plugin | [Github.Plugin]): Github
 
   registerEndpoints(routes: any): void
 

--- a/test/typescript-validate.ts
+++ b/test/typescript-validate.ts
@@ -77,7 +77,7 @@ export default async function() {
     // add logic before, after, catch errors or replace the request altogether
     return request(options)
   })
-  octokit.plugin((octokit, options) => {
+  Octokit.plugin((octokit, options) => {
     octokit.hook.wrap('request', async (request, options) => {
       const time = Date.now()
       const response = await request(options)

